### PR TITLE
driver: crypto: hisilicon: fix error handling

### PR DESCRIPTION
**Pull request by yuzexiyzx**
1. support num = 0 in qm_set_vt_common().
2. release qp when qm_xqc_cfg() failed.
3. add check of msg and qp->num in hisi_qp_recv_sync().

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
There's a typo in the title of the commit, I guess it should be "exception".

Please explain in plain text in the commit message why you're doing this. Most of what you're doing can be seen in the diff, but sometimes it's nice with a summary of that too.

**Comment by yuzexiyzx** [comment]
@jenswi-linaro comment solved.
changed to
1. support num = 0 in qm_set_vft_common().
When qm_set_xqc_set() failed, it will set q_num to zero.
qm_set_vft_common() need to support this scene.
2. release qp when qm_xqc_cfg() failed.
3. add check of msg and qp->num in hisi_qp_recv_sync().
hisi_qp_recv_sync() is not static function, we need to
check every input parameter to ensure that it is safe.

**Comment by jenswi-linaro** [comment]
I'm sorry I think you missed my point. You're in principle describing the diff in detail. In a good commit message you **describe why a change is being made, not just what's changing**.

If you need to make a list of all things done in a patch it's usually a sign that it should have been multiple patches, but in this case, it seems that it all belongs in one patch.

How about something like:
```
driver: crypto: hisilicon: fix error handling

When the XXX fails to configure, qm_set_xqc_vft() is called with the num argument
as zero to disable the device (or whatever happens). Update qm_set_xqc_vft() to
handle this error path.
```

File: core/drivers/crypto/hisilicon/hisi_qm.c [hunk 1]
```diff
@@ -736,18 +731,20 @@ struct hisi_qp *hisi_qm_create_qp(struct hisi_qm *qm, uint8_t sq_type)
...
 		EMSG("Fail to set qp[%"PRIu32"] cqc", qp->qp_id);
-		goto err_proc;
+		goto qp_release;
 	}
 
 	qm->qp_in_used++;
 	mutex_unlock(&qm->qp_lock);
 	return qp;
 
+qp_release:
```
[the remark this answers is not in the document]
Comment (yuzexiyzx) [reply core/drivers/crypto/hisilicon/hisi_qm.c]: solved

**Comment by yuzexiyzx** [comment]
@jenswi-linaro @etienne-lms All comments have been solved.

**Comment by jenswi-linaro** [comment]
The title of the commit says:
> driver: crypto: hisilicon: fix exception handling 

I think the word "exception" might be misleading since we're not dealing with what we normally call exceptions in this patch. How about using the word "error" instead?

**Comment by jenswi-linaro** [comment]
With my comment addressed, please apply:
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by yuzexiyzx** [comment]
@jforissier All comments have been solved.

**Comment by jforissier** [comment]
The IBART error can be ignored (rebase needed). @yuzexiyzx thanks!

**Pull request merged**